shell_elvish.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"strings"
+)
+
+type elvish struct {
+	re *strings.Replacer
+}
+
+var ELVISH = elvish{}
+
+func (elvish) Hook() (string, error) {
+	return `fn -direnv-hook {
+	direnv export elvish > .direnv.elv
+	-source .direnv.elv
+	/bin/rm .direnv.elv
+}
+
+edit:before-readline = [{ -direnv-hook }]
```

Comment:
Do not overwrite the entire hook list. Use:

```
edit:before-readline = [$@edit:before-readline { -direnv-hook }]
```

or equivalently

```
@edit:before-readline = $@edit:before-readline { -direnv-hook }
```

You can also inline the hook function in here.